tiff-2-png-converter.py

```diff
@@ -1,197 +1,358 @@
+                    tmp_img = cv2.resize(img, resolution_image)
+                    tmp_img[:, :] = 0
+                if j == 0:
+                    imgs_row = np.copy(tmp_img)
                 else:
-                    print("image " +file_full_path + "has a different size " + "("+str(width)+","+str(height)+")")
-                    sys.exit()
-    else:
-        print("Folder does not exists")
+                    imgs_row = cv2.hconcat([imgs_row, tmp_img])
+                count = count + 1
+            if i == 0:
+                image_out = np.copy(imgs_row)
+            else:
+                image_out = cv2.vconcat([image_out, imgs_row])
+        print_progress_bar(
+            i + 1, dim, prefix="Mapping slices to 2D:", suffix="Complete", length=50
+        )
+
+        name, extension = os.path.splitext(self.dest)
+        if not extension:
+            extension = ".png"
+
+        amax = np.amax(image_out)
+        norm = np.zeros((image_out.shape))
+        normalized_image_u16bit = cv2.normalize(
+            image_out, norm, 0, 2 ** 16, cv2.NORM_MINMAX
+        )
+        numpngw.write_png(name + extension, image_out)
+        numpngw.write_png(name + "_normalized" + extension, normalized_image_u16bit)
+
+        file = open(name + "_metadata", "a")
+        file.write("Width:" + str(width) + "\n")
+        file.write("Heigth:" + str(height) + "\n")
+        file.write("Depth:" + str(self.num_imgs) + "\n")
+        file.write("bps:" + str(image_out.dtype) + "\n")
+        file.write("amax:" + str(amax) + "\n")
+        file.close()
+        print("Image written to file-system : " + name + extension)
+
+    def merge_RGB(
+        self, image_r, image_g, image_b, rgb_images, width, heigh, bits_per_sample
+    ):
+        zero_image = np.zeros((width, heigh), dtype=bits_per_sample)
+        print_progress_bar(
+            0, self.num_imgs, prefix="Mergin channels:", suffix="Complete", length=50
+        )
+        for z in range(self.num_imgs):
+            r_channel = np.zeros((width, heigh), dtype=bits_per_sample)
+            g_channel = np.zeros((width, heigh), dtype=bits_per_sample)
+            b_channel = np.zeros((width, heigh), dtype=bits_per_sample)
+            if image_r[z] is not None:
+                r_channel = image_r[z]
+                if self.equalize_histogram:
+                    r_channel = self.equalize_histogram_method(r_channel)
+            else:
+                r_channel = zero_image
+            if image_g[z] is not None:
+                g_channel = image_g[z]
+                if self.equalize_histogram:
+                    g_channel = self.equalize_histogram_method(g_channel)
+            else:
+                g_channel = zero_image
+            if image_b[z] is not None:
+                b_channel = image_b[z]
+                if self.equalize_histogram:
+                    b_channel = self.equalize_histogram_method(b_channel)
+            else:
+                b_channel = zero_image
```

Comment:
this chunk is pretty repetitive, could it be pulled out into a function? I wonder if passing around the three r/g/b images as a list of [r, g, b] might simplify some code

Reply:
I reworked the function.